sockeye/utils.py

```diff
@@ -610,7 +610,12 @@ def __init__(self, candidates: List[int], lock_dir: str) -> None:
     def __enter__(self) -> Optional[int]:
         for gpu_id in self.candidates:
             lockfile_path = os.path.join(self.lock_dir, "sockeye.gpu%d.lock" % gpu_id)
-            lock_file = open(lockfile_path, 'w')
+            try:
+                lock_file = open(lockfile_path, 'w')
+            except IOError as e:
+                if errno.EACCES:
+                    logger.warning("GPU %d is currently locked by a different process (Permission denied).", gpu_id)
+                    return None
```

Comment:
Is it better to continue the for loop and try to find another available GPU? The same for L641.

Reply:
this is what effectively happens. It just happens outside the class in this while loop: https://github.com/awslabs/sockeye/blob/master/sockeye/utils.py#L563

Reply:
(where we check whether `None` was returned)

Reply:
Got it. Thanks!